### PR TITLE
Update rules metadata, dependencies, copyright and remove simple code smells

### PR DESCRIPTION
**Pull request by francoismora**

### CODE REVIEW

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java [hunk 1]
```diff
@@ -33,13 +34,19 @@ public final class FlexRulesDefinition implements RulesDefinition {
   private static final String RESOURCE_BASE_PATH = "org/sonar/l10n/flex/rules/flex";
   private static final Set<String> TEMPLATE_RULE_KEYS = new HashSet<>(Arrays.asList("XPath", "CommentRegularExpression"));
 
+  private final SonarRuntime sonarRuntime;
```
Comment (saberduck) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java:37]: can we move this change into another commit?

File: sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java [hunk 2]
```diff
@@ -20,22 +20,27 @@
 package org.sonar.plugins.flex;
 
 import org.junit.Test;
+import org.sonar.api.SonarRuntime;
```
Comment (saberduck) [inline sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java:23]: should not be part of this commit

File: sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexSquidSensorTest.java [hunk 3]
```diff
@@ -28,6 +28,9 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.mockito.Mockito;
+import org.mockito.internal.creation.MockSettingsImpl;
```
Comment (saberduck) [inline sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexSquidSensorTest.java:31]: should not be part of this commit

File: flex-checks/src/main/java/org/sonar/flex/checks/XPathCheck.java [hunk 4]
```diff
@@ -76,7 +76,7 @@ public void visitFile(@Nullable AstNode fileNode) {
         if (object instanceof AstNode) {
           AstNode astNode = (AstNode) object;
           addIssueAtLine(message, astNode.getTokenLine());
-        } else if (object instanceof Boolean && (Boolean) object) {
+        } else if (object instanceof Boolean && Boolean.TRUE.equals(object)) {
```
Comment (saberduck) [inline flex-checks/src/main/java/org/sonar/flex/checks/XPathCheck.java:79]: `instanceof` is not needed, equals will cover it

**Review by saberduck — APPROVED** [review]
some minor comments, otherwise lgtm!

File: sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java (hunk 2 above)
Comment (francoismora) [reply sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java:23]: This change is brought by the latest version of the API that requires this additional SonarRuntime parameter to load the rules metadata (see `FlexRulesDefinition.define(Context context)`)

File: sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexSquidSensorTest.java (hunk 3 above)
Comment (francoismora) [reply sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexSquidSensorTest.java]: Moved to the commit where versions are updated

File: sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java (hunk 2 above)
Comment (francoismora) [reply sonar-flex-plugin/src/test/java/org/sonar/plugins/flex/FlexRulesDefinitionTest.java:23]: Moved to the commit where versions are updated

File: flex-checks/src/main/java/org/sonar/flex/checks/XPathCheck.java (hunk 4 above)
Comment (francoismora) [reply flex-checks/src/main/java/org/sonar/flex/checks/XPathCheck.java]: Changed
